File: Data/Java/prime checker.java
public class PrimeChecker {

    // Function to check if a number is prime
    public static boolean isPrime(int n) {
        if (n <= 1) return false;  // Not prime
        if (n == 2) return true;   // 2 is prime

        // Check for divisibility from 2 to sqrt(n)
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0)
                return false;  // Divisible => not prime
        }

        return true;  // No divisors => prime
    }

    public static void main(String[] args) {
        int n = 29;

        boolean result = isPrime(n);

        System.out.println(n + " is prime? " + result);  // Output: true
    }
}

File: Data/Python/Prime checker.py
def is_prime(n):
    """
    Prime Checker:
    - A prime number has exactly 2 divisors: 1 and itself.
    - Check for divisibility up to sqrt(n).
    - Time Complexity: O(√n)
    """
    if n <= 1:
        return False  # 0 and 1 are not prime
    if n <= 3:
        return True   # 2 and 3 are prime

    if n % 2 == 0 or n % 3 == 0:
        return False  # Eliminate even numbers and multiples of 3

    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False  # Check divisibility by i and i+2
        i += 6
    return True

# ✅ Example
print("Is 29 Prime?:", is_prime(29))  # Output: True
print("Is 40 Prime?:", is_prime(40))  # Output: False
